### ISSUE TITLE
More options for the Sound extension

### ISSUE THREAD

**Issue by Alestore** [TurboWarp/extensions#466]
I think the Sound extension has a lot of potential.
I haven't tried yet if it's possible to start the sound from a local directory (presumably from the same folder as the project).

What I would definitely like to see in this extension is:
- Ability to stop a sound while it's playing
- Blocks to change sound volume
- Apply effects (similar to the blocks included in Scratch related to Sound)

This extension, with the additions that I have recommended, would be a solution to the continuous lag and the heaviness of certain projects that have loaded the entire running project into RAM.
Making the sounds available within a local directory would allow the audio to play even offline and without sound load times (if heavy enough).
Also, playing audio files only when needed without having all other audio loaded into RAM memory (even when they are not being played), would save a lot of memory and the initial load times of packaged projects would be drastically reduced.

**Comment by Alestore** [comment]
> I think there should also be a block that plays a sound until it's stopped so you don't have to stop all scripts in the sprite or add a variable to stop a sound that's in a forever loop. Also the block that you mentioned which stops a sound while it's playing should have a dropdown box to stop sounds from specific sprites.

There is already a block that says "Play audio from (location) until done". Maybe this is what you are suggesting.

I completely agree with the drop-down box to stop specific sounds from specific sprites, this sounds really useful.

**Comment by CST1229** [comment]
> By "Play sound until stopped" I mean for it to play a sound until your suggested stop a sound while its playing block is used.

Would it loop the sound until the stop block is used?

**Comment by gomiigg** [comment]
> Ability to stop a sound while it's playing

You can do this in base Turbowarp already. Turn off Misc. Limits, then `set pitch effect to (-1 / empty value)`. The result is  a pitch of -Infinity which basically slows down the audio so much that it's practically silent and paused.

**Comment by Alestore** [comment]
> > Ability to stop a sound while it's playing
> 
> You can do this in base Turbowarp already. Turn off `Misc. Limits` (disables sound limits), then `set pitch effect to (-1 / empty value)`. The result is  a pitch of `-Infinity` which basically slows down the audio so much that it's practically silent and paused.

Yes ok but if I want to put another sound from the same Sprite when the previous sound is not finished and is only paused, putting back the normal pitch they will both continue mixing

**Comment by Alestore** [comment]
> > Ability to stop a sound while it's playing
> 
> You can do this in base Turbowarp already. Turn off `Misc. Limits` (disables sound limits), then `set pitch effect to (-1 / empty value)`. The result is  a pitch of `-Infinity` which basically slows down the audio so much that it's practically silent and paused.

I'd rather like a block like "stop all sounds" (like in basic TurboWarp)

**Comment by DNin01** [comment]
I've suggested a couple of these features before: #114

**Comment by LukeManiaStudioss** [comment]
> I've suggested a couple of these features before: #114

This refers specifically to the Sound extension that allows you to play sounds from URLs

**Comment by SharkPool-SP** [comment]
Try this, It has everything youre looking for, and more!
**Tune Shark V3**
You can find it here:
**https://sharkpools-extensions.vercel.app/**
or
**https://github.com/TurboWarp/extensions/pull/1489**

Load it _unsandboxed_
